detect_secrets/plugins/common/filetype.py

```diff
@@ -18,22 +41,8 @@ def determine_file_type(filename):
 
     :rtype: FileType
     """
-    if filename.endswith('.cls'):
-        return FileType.CLS
-    elif filename.endswith('.go'):
-        return FileType.GO
-    elif filename.endswith('.java'):
-        return FileType.JAVA
-    elif filename.endswith('.js'):
-        return FileType.JAVASCRIPT
-    elif filename.endswith('.php'):
-        return FileType.PHP
-    elif filename.endswith('.py'):
-        return FileType.PYTHON
-    elif (
-        filename.endswith(
-            ('.eyaml', '.yaml', '.yml'),
-        )
-    ):
-        return FileType.YAML
-    return FileType.OTHER
+    _, file_extension = os.path.splitext(filename)
```

Comment:
Does this handle `my.weird.filename`?

Reply:
Yup
```python
>>> os.path.splitext('my.weirdo.filename')
('my.weirdo', '.filename')
```

Reply:
Oh right, and does this work with no file extension?

Reply:
Oh it does, the `get` call in the next line has a default

Reply:
For posterity, it'll be `''` and then get set to `FileType.OTHER`

(I had an internal 🤔 about `setdefault` vs. the `.get` fallback but 🤷‍♂ I kinda liked the way it looked.)